### ISSUE TITLE
How to set IMAGE_MEAN and IMAGE_STD

### ISSUE THREAD

**Issue by JohnTian** [googlecodelabs/tensorflow-for-poets-2#2]
Firstly, thanks very much for your work on [tensorflow-for-poets-2](https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2/#0).

I followed your post in  [tensorflow-for-poets-2](https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2/#0).

Although, the result of my own 25 categories is 99.9% in Ubuntu 16.04. The result of the model in android is very bad.

I think IMAGE_MEAN and IMAGE_STD may cause it. So I'm confused how to set them.
Any suggestion is appreciated.

**Comment by MarkDaoust** [comment]
Hi, 

99.9 sounds a little high ... are you cross validating to make sure you aren't overfitting your data?

IMAGE_MEAN and IMAGE_STD aren't meant to be a function of the *images*. They're more a question of the expected input range of the the network being used. Some nets want the input on the [0,255] range, others [0,1],  others [-1,1],... and having these "mean" and "variance" parameters to do what looks like a Z-score normalization (subtract the mean, and divide by the variance) is a simple way to enforce that.

I really don't think they're the cause of your problem.

Good luck!

**Comment by MarkDaoust** [comment]
If you really want to try, the values are right there in [ClassifierActivity.java](https://github.com/googlecodelabs/tensorflow-for-poets-2/blob/01e5a68183d158c5b04944e8f44674eeaa91f679/android/src/org/tensorflow/demo/ClassifierActivity.java#L61) but you should not have to update these, I'm prrety sure it can only make things worse.

**Comment by JohnTian** [comment]
@MarkDaoust Thanks a lot. I just followed the steps post in [tensorflow-for-poets-2](https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2/#0). I will do cross validating to make sure whether the model overfitting or not.
